### PR TITLE
xds: add buildXdsCluster test

### PR DESCRIPTION
**Pull request by cnvergence**
- https://github.com/envoyproxy/gateway/issues/1078

### CODE REVIEW

File: internal/xds/translator/cluster_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	bootstrapStr, err := bootstrap.GetRenderedBootstrapConfig()
+	require.NoError(t, err)
+	jsonData, err := yaml.YAMLToJSON([]byte(bootstrapStr))
+	require.NoError(t, err)
+	err = protojson.Unmarshal(jsonData, bootstrapObj)
+	require.NoError(t, err)
+
+	for _, cluster := range bootstrapObj.StaticResources.Clusters {
+		if cluster.Name == xdsClusterName {
+			bootstrapXdsCluster = cluster
```
Comment (arkodg) [inline internal/xds/translator/cluster_test.go:62]: nit: you can return directly here
```
return cluster
```

and you can fail the test if it reaches the end of the func

**Review by arkodg — DISMISSED** [review]
LGTM thanks !
will help us catch any config drifts in the future

**Comment by arkodg** [comment]
seeing a lint error in the CI
```
Error: File is not `goimports`-ed with -local github.com/envoyproxy/gateway/ (goimports)
[114](https://github.com/envoyproxy/gateway/actions/runs/4620634485/jobs/8173016266?pr=1257#step:5:115)
make[1]: *** [tools/make/lint.mk:23: lint.golint] Error 1
```

**Review by arkodg — APPROVED** [review]

**Review by AliceProxy — APPROVED** [review]
lgtm, thanks :slightly_smiling_face:

**Pull request merged**